File: packages/transform/src/features/json/JsonAssertStringifyTransformer.ts
import { JsonAssertStringifyProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonAssertStringifyTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "json.assertStringify",
      write: JsonAssertStringifyProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonAssertStringifyTransformer.go
package json

import (
	corejson "github.com/samchon/typia/packages/core/src/programmers/json"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonAssertStringifyTransformer = jsonAssertStringifyTransformerNamespace{}

type jsonAssertStringifyTransformerNamespace struct{}

func (jsonAssertStringifyTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corejson.JsonStringifyProgrammer.Write(call.TypeText)
}
